cmd/pbackend/statuser.go

```diff
@@ -153,16 +153,24 @@ func checkSourceAvailabilityAWS(ctx context.Context) {
 				Identity:     s.Identity,
 				ResourceType: "Application",
 			}
-			_, err = clients.GetEC2Client(ctx, &s.Authentication, "")
+			ec2Client, err := clients.GetEC2Client(ctx, &s.Authentication, "")
 			if err != nil {
 				sr.Status = kafka.StatusUnavailable
 				sr.Err = err
 				logger.Warn().Err(err).Msg("Could not get aws assumed client")
-				chSend <- sr
 			} else {
-				sr.Status = kafka.StatusAvaliable
-				chSend <- sr
+				sr.Status = kafka.StatusAvailable
 			}
+			permissions, err := ec2Client.CheckPermission(ctx, &s.Authentication)
+			if err != nil {
+				sr.Err = err
```

Comment:
Here you should also set `sr.Status = kafka.StatusUnavailable` because missing permissions effectively means the source is not "all right" and the user should delete it and create it again.